packages/nuxt/src/pages/utils.ts

```diff
@@ -88,11 +88,15 @@ export function generateRoutesFromFiles (files: string[], pagesDir: string): Nux
     }
 
     parent.push(route)
+    // TODO: https://github.com/vuejs/router/issues/1435
+    parent.sort((a, b) => getSortablePath(a.path).localeCompare(getSortablePath(b.path)))
   }
 
   return prepareRoutes(routes)
 }
 
+const getSortablePath = (path: string) => path.replace(/^\//, '').replace(/:/, 'z')
```

Comment:
Z is after z. At least a better workaround with localeCompare 
```suggestion
const getSortablePath = (path: string) => path.replace(/^\//, '').replace(/:/, 'Z')
```